client/src/components/main/CreateNewPost.jsx

```diff
@@ -20,14 +22,40 @@ import {
   handleLocationInput
 } from '../../store/modules/newpost';
 
+
+
 const CreateNewPost = (props) => {
 
+
+  const geocodeLocationInput = (location) => {
+    let url = `https://maps.googleapis.com/maps/api/geocode/json?address=${location}&key=AIzaSyDXLOMgs19AOUHeizaMnRwjVyzxcTGWmJ8`
+    return axios.get(url)
+    .then((res) => {
+      console.log('response from geocoding API: ', res);
+      props.handleLocationInput(res.data.results[0].geometry.location);
+    })
+    .catch((err) => {
+      console.log('(Client) Error calling Google Geocoding API');
+    });
+  }
+
+  // Autocomplete feature for the form's location input field
+  const initLocationAutocomplete = () => {
+    let input = document.getElementById('locationInput');    
+    let autocomplete = new google.maps.places.Autocomplete(input);
+    let place;
+    google.maps.event.addListener(autocomplete, 'place_changed', () => {
+      place = autocomplete.getPlace();
+      geocodeLocationInput(place.formatted_address);
+    });
```

Comment:
Can we think about changing this variable name?

Reply:
Can we also comment this? It's confusing.

Reply:
Yeah definitely. Will add comments and rename function. Thanks.